Tokencache should handle invalid keys well

Correct test shows the issue:

```ts
it.only('should reject if there is no token configuration for given name', () => {

      // given
      nock(oauthHost)
        .post('/access_token')
        .reply(HttpStatus.OK, {
          access_token: defaultAccessTokenValue
        })
        .get('/tokeninfo')
        .query({ access_token: defaultAccessTokenValue })
        .reply(HttpStatus.OK, defaultTokenInfoResponse);

      // when
      const tokenCache = new TokenCache({
        'nucleus': ['nucleus.write', 'nucleus.read'],
        'halo': ['all']
      }, oauthConfig);

      // then
      return expect(tokenCache.get('foo')).to.be.rejected;
    });
```

Current test response 

```
1) TokenCache
       get
         should reject if there is no token configuration for given name:
     AssertionError: expected promise to be rejected but it was fulfilled with { Object (expires_in, token_type, ...) }
```

--- a/integration-test/oauth-tooling/tokenCache.spec.ts
+++ b/integration-test/oauth-tooling/tokenCache.spec.ts
@@ -50,7 +50,12 @@ describe('TokenCache', () => {
       // given
       nock(oauthHost)
         .post('/access_token')
-        .reply(HttpStatus.INTERNAL_SERVER_ERROR);
+        .reply(HttpStatus.OK, {
+          access_token: defaultAccessTokenValue
+        })
+        .get('/tokeninfo')
+        .query({ access_token: defaultAccessTokenValue })
+        .reply(HttpStatus.OK, defaultTokenInfoResponse);
 
       // when
       const tokenCache = new TokenCache({
@@ -59,7 +64,7 @@ describe('TokenCache', () => {
       }, oauthConfig);
 
       // then
-      return expect(tokenCache.get('foo')).to.be.rejected;
+      return expect(tokenCache.get('foo')).to.be.eventually.rejectedWith('TokenCache miss: foo does not exist');
     });
 
     it('should reject if there is no token and is not able to request a new one', () => {

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "authmosphere",
-  "version": "2.0.2-rc9",
+  "version": "2.0.2-rc10",
   "description": "authmosphere is a library to support OAuth2 workflows in JavaScript projects.",
   "main": "./lib/src/index.js",
   "typings": "./lib/src/index.d.ts",

--- a/src/token-cache.ts
+++ b/src/token-cache.ts
@@ -90,7 +90,14 @@ class TokenCache {
    */
   get(tokenName: string): Promise<Token> {
 
-    const promise = this.validateToken(tokenName)
+    if (!this.isTokenConfigured(tokenName)) {
+      const msg = `TokenCache miss: ${tokenName} does not exist`;
+      this.logger.debug(msg);
+
+      return Promise.reject(msg);
+    }
+
+    const promise = this.getCachedToken(tokenName)
       .catch(() => {
 
         const config = {
@@ -155,6 +162,10 @@ class TokenCache {
       .then(() => this._tokens);
   }
 
+  private isTokenConfigured(tokenName: string): boolean {
+    return this.tokenConfig[tokenName] !== undefined;
+  }
+
   /**
    * Checks whether a valid token for the given name is present.
    * Resolves with that token if that is the case.
@@ -163,13 +174,7 @@ class TokenCache {
    * @param tokenName
    * @returns {Promise<Token>}
    */
-  private validateToken(tokenName: string): Promise<Token> {
-
-    if (this.tokenConfig[tokenName] === undefined) {
-      const msg = `TokenCache miss: ${tokenName} does not exist`;
-      this.logger.error(msg);
-      return Promise.reject(msg);
-    }
+  private getCachedToken(tokenName: string): Promise<Token> {
 
     const token = this._tokens[tokenName];
 

